scripts/appleseedMaya/renderGlobals.py

```diff
@@ -405,20 +406,24 @@ def create(self):
                         adaptiveSampling = mc.getAttr("appleseedRenderGlobals.adaptiveSampling")
 
                         self._addControl(
-                            ui=pm.intFieldGrp(
-                                label="Min Samples", numberOfFields=1, enable=adaptiveSampling),
+                            ui=pm.intSliderGrp(
+                                label="Min Samples", field=True, value=16, cw=(3,160), minValue=0, 
+                                fieldMinValue=0, maxValue=256, fieldMaxValue=1000000, enable=adaptiveSampling),
                             attrName="minPixelSamples")
                         self._addControl(
-                            ui=pm.intFieldGrp(
-                                label="Max Samples", numberOfFields=1),
+                            ui=pm.intSliderGrp(
+                                label="Max Samples", field=True, value=256, cw=(3,160), minValue=16, 
```

Comment:
Hard min value for Max Samples should be 0 as in appleseed.studio. Whether 0 is a reasonable hard min value or not is debatable, but at least that would be consistent with .studio.

Reply:
Usually the hard limits are set in the renderglobalsnode.cpp where the parameter are defined.
Here are the input limits for the UI elements (sliders and numeric fields). 
maxValue and minValue are the soft limits for the slider ranges, so a user can see the useful parameter range from the slider range. 
fieldMaxValue and fieldMinValue are the input limits for the numeric fields and override the slider ranges. If a user wants to choose a value outside of the slider range he can just type it then into the numeric field and override the slider (which adapts then to the new input).
If no fieldMaxValue or fieldMinValue are specified here then they are automatically equal to the hard limits of the parameter definition in the c++ code.